crates/uv-resolver/src/pubgrub/report.rs

```diff
@@ -1066,6 +1075,88 @@ impl PackageRange<'_> {
     }
 }
 
+/// Create a range with improved segments for reporting the available versions for a package.
+fn update_availability_range(
+    range: &Range<Version>,
+    available_versions: &BTreeSet<Version>,
+) -> Range<Version> {
+    let mut new_range = Range::empty();
+
+    let (available_range, first_available, last_available) =
+        match (available_versions.first(), available_versions.last()) {
+            // At least one version is available
+            (Some(first), Some(last)) => {
+                let range = Range::<Version>::from_range_bounds((
+                    Bound::Included(first.clone()),
+                    Bound::Included(last.clone()),
+                ));
+                // If only one version is available, return this as the bound immediately
+                if first == last {
+                    return range;
+                }
+                (range, first, last)
+            }
+            // SAFETY: If there's only a single item, `first` and `last` should both
+            // return `Some`.
+            (Some(_), None) | (None, Some(_)) => unreachable!(),
+            // No versions are available; nothing to do
+            (None, None) => return Range::empty(),
+        };
+
+    for segment in range.iter() {
+        let (lower, upper) = segment;
+        let segment_range = Range::from_range_bounds((lower.clone(), upper.clone()));
+
+        // Drop the segment if it's disjoint with the available range, e.g., if the segment is
+        // `foo>999`, and the the available versions are all `<10` it's useless to show.
+        if segment_range.is_disjoint(&available_range) {
+            continue;
+        }
```

Comment:
What if the range is a singleton, and it's not in available versions?

Reply:
(Is that impossible?)

Reply:
Like this?

```
❯ uv add 'httpx==9999'
  ╰─▶ Because there is no version of httpx==9999 and example depends on httpx==9999, we can conclude that example's requirements
      are unsatisfiable.
      And because your workspace requires example, we can conclude that your workspace's requirements are unsatisfiable.
```

I think that's always handled before we get here. Otherwise, this operation is performed on the complement so I don't think it's possible.

Reply:
Yeah like that. Ok I figured it was somehow handled upstream.